source/common/rds/basic/route_config_provider_manager_impl.h

```diff
@@ -0,0 +1,89 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/singleton/instance.h"
+
+#include "source/common/config/opaque_resource_decoder_impl.h"
+#include "source/common/config/resource_name.h"
+#include "source/common/rds/basic/config_traits_impl.h"
+#include "source/common/rds/basic/proto_traits_impl.h"
+#include "source/common/rds/basic/route_config_provider_manager.h"
+#include "source/common/rds/rds_route_config_provider_impl.h"
+#include "source/common/rds/rds_route_config_subscription.h"
+#include "source/common/rds/route_config_provider_manager.h"
+#include "source/common/rds/route_config_update_receiver_impl.h"
+#include "source/common/rds/static_route_config_provider_impl.h"
+
+#include "absl/strings/ascii.h"
+
+namespace Envoy {
+namespace Rds {
+namespace Basic {
+
+/**
+ * Implementation of RouteConfigProviderManager interface.
+ * The keys for config tracker, stat, log and exception text will be derived
+ * from the name of the proto message class passed in the Rds template argument
+ * Since the config tracker key has to be unique across envoy, this name has to be also unique.
+ * The proto message has two mandatory fields:
+ * config_source (config.core.v3.ConfigSource)
+ * route_config_name (string)
```

Comment:
If `route_config_name` is mandatory, should the corresponding field in the `Trds` service be required?

Reply:
The mandatory here means this field must be declared in the proto message since the template will call `rds.route_config_name()` but the content is not checked so it can be undefined.
I make the comment more clear.